frontend/src/components/utils/ApproveLanguageModal.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from "react";
+import Select, { StylesConfig } from "react-select";
+import {
+  Button,
+  Flex,
+  Heading,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Text,
+  Box,
+  Grid,
+  GridItem,
+  useStyleConfig,
+} from "@chakra-ui/react";
+
+import { languageOptions } from "../../constants/Languages";
+import { levelOptions } from "../../constants/Levels";
+import { roleOptions } from "../../constants/Roles";
+
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+import { ApprovedLanguagesMap } from "../../utils/Utils";
+
+export type StoryToAssign = {
+  storyId: number;
+  storyTranslationId?: number;
+  title: string;
+  description: string;
+  level: number;
+  language: string;
+};
+
+export type NewApprovedLanguage = {
+  language: string;
+  level: number;
+  role: string;
+};
+
+export type ApprovedLanguageModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onAssign: (newApprovedLanguage: NewApprovedLanguage) => void;
+  approvedLanguagesTranslation: ApprovedLanguagesMap;
+  approvedLanguagesReview: ApprovedLanguagesMap;
+};
+
+const DropdownIndicator = () => {
+  return (
+    <Box
+      borderLeft="5px solid transparent"
+      borderRight="5px solid transparent"
+      borderTop="9px solid black"
+      margin="10px"
+    />
+  );
+};
+
+const colourStyles: StylesConfig = {
```

Comment:
Looks like this code also lives in `TableFilter` and `AssignStoryModal`, mind extracting this?